sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/OsResource.java

```diff
@@ -7,25 +7,33 @@
 
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.common.AttributesBuilder;
-import io.opentelemetry.sdk.resources.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
 import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
 import javax.annotation.Nullable;
 
-/** {@link ResourceProvider} which provides information about the current operating system. */
-public final class OsResource extends ResourceProvider {
+/** {@link Resource} which provides information about the current operating system. */
```

Comment:
Technically, this class isn't a resource, it just contains a method that returns one.

How about sth like
```suggestion
/** Provides a {@link Resource} containing information about the current operating system. */
```
instead?

Reply:
Yeah it was a bit weird - went with factory to use a noun like is usual for class javadoc. With the change, I liked the `getnstance` name even less than I did before so renamed it to just `get`